ci/windows-pkg-arrow-for-r.sh

```diff
@@ -54,6 +54,10 @@ wget https://dl.bintray.com/rtools/backports/mingw-w64-x86_64-snappy-1.1.7-2-any
 # double-conversion is only available in the Rtools4.0 builds, but apparently that's ok
 wget https://dl.bintray.com/rtools/mingw64/mingw-w64-x86_64-double-conversion-3.1.2-1-any.pkg.tar.xz
 wget https://dl.bintray.com/rtools/mingw32/mingw-w64-i686-double-conversion-3.1.2-1-any.pkg.tar.xz
+
+wget https://dl.bintray.com/rtools/mingw64/:mingw-w64-x86_64-openssl-1.1.1.a-1-any.pkg.tar.xz
+wget https://dl.bintray.com/rtools/mingw32/:mingw-w64-i686-openssl-1.1.1.a-1-any.pkg.tar.xz
```

Comment:
CC: @nealrichardson

Reply:
Thanks, that seems ok (though you need to remove the extra `:` in the middle of the URLs).

Since you had to make this change, I suspect that this means we'll need to update the Homebrew formula too, but unfortunately we don't have that under CI.

Reply:
Fixed the URLs. What are your thoughts on updating the Homebrew formula?

Reply:
IMO we should put the C++ homebrew formula in our `ci/` directory and have a Travis job, perhaps only run nightly/on-demand if we're concerned about build time, that builds from it and then runs some test suite that requires it (e.g. R, Ruby). https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md suggests we can `brew install --build-from-source` so IIUC we'd use something like that. 

Related, I previously tried to add `brew install --HEAD` to CI but that didn't go anywhere: https://github.com/apache/arrow/pull/4576

I don't think this needs to block this patch, but it would be great to do something with this sooner rather than later. Otherwise we'll keep making releases unnecessarily expensive and that will keep us from doing them as frequently as we should.